docs/examples/spectrum/spectral_factor.py

```diff
@@ -0,0 +1,155 @@
+# More information on a range of spectral factor models, as well as the
+# variables upon which they are based, can be found in [1]_.
+#
+# Let's import some data. This example uses a Typical Meteorological Year 3
+# (TMY3) file for the location of Greensboro, North Carolina, from the pvlib
+# data directory. This TMY3 file is constructed using the median month from
+# each year between 1980 and 2003, from which we extract the first week of
+# August 2001 to analyse.
+
+# %%
+import pathlib
+from matplotlib import pyplot as plt
+import pandas as pd
+import pvlib as pv
+from pvlib import location
+from pvlib.solarposition import get_solarposition
+from pvlib.atmosphere import get_relative_airmass
+
+DATA_DIR = pathlib.Path(pv.__file__).parent / 'data'
+meteo, metadata = pv.iotools.read_tmy3(DATA_DIR / '723170TYA.CSV',
+                                       map_variables=True)
+meteo = meteo.sort_index()
+# meteo = meteo.between_time('06:00', '20:00').loc['2001-08-01':'2001-08-07']
+meteo = meteo.loc['2001-08-01':'2001-08-07']
+
+# %%
+# pvlib Spectral Factor Functions
+# -----------------------------
+# This example demonstrates the application of three pvlib spectral factor
+# functions, namely `SAPM`_, `PVSPEC`_, and `firstsolar`_.
+
+# %%
+# SAPM
+# ^^^^
+# The SAPM function (:py:func:`pvlib.spectrum.spectral_factor_sapm`) for the
+# spectral mismatch factor calculates :math:`M` using absolute airmass,
+# :math:`AM_a` as an input.
+
+# %%
+# PVSPEC
+# ^^^^^^
+# The PVSPEC function (:py:func:`pvlib.spectrum.spectral_factor_pvspec`) for
+# the spectral mismatch factor calculates :math:`M` using :math:`AM_a` and the
+# clearsky index, :math:`k_c` as inputs.
+
+# %%
+# First Solar
+# ^^^^^^^^^^^
+# The First Solar function
+# (:py:func:`pvlib.spectrum.spectral_factor_firstsolar`) for the spectral
+# mismatch factor calculates :math:`M` using :math:`AM_a` and the atmospheric
+# precipitable water content, :math:`W`, as inputs.
+
+# %%
+# Calculation of inputs
+# ---------------------
+# Let's calculate the absolute air mass, which is required for all three
+# models. We use the Kasten and Young [2]_ model, which requires the apparent
+# sun zenith. Note: TMY3 files timestamps indicate the end of the hour, so we
+# shift the indices back 30-minutes to calculate solar position at the centre
+# of the interval.
+
+# Create a location object
+lat, lon = metadata['latitude'], metadata['longitude']
+alt = altitude = metadata['altitude']
+tz = 'Etc/GMT+5'
+loc = location.Location(lat, lon, tz=tz, name='Greensboro, NC')
+
+# Calculate solar position parameters
+solpos = get_solarposition(
+    meteo.index.shift(freq="-30min"), lat, lon, alt,
+    pressure=meteo.pressure*100,  # convert from millibar to Pa
+    temperature=meteo.temp_air)
+solpos.index = meteo.index  # reset index to end of the hour
+
+amr = get_relative_airmass(solpos.apparent_zenith).dropna()  # relative airmass
+ama = amr*(meteo.pressure/1013.25)  # absolute airmass
```

Comment:
`pvlib.atmosphere.get_absolute_airmass`?

Reply:
I could, because I have already imported pvlib, but I wrote it like this since I included statements such as `from pvlib.atmosphere import get_relative_airmass` at the start. I sometimes mix between using and not using from X import X a bit erratically to be honest... not sure what is best python practice.
Do you advise getting rid of those additional import statements and just writing it out in full as you have commented here?

Reply:
I had that question too and the suggestion from @AdamRJensen was to import the general package (e.g., `pvlib`) and then call each function in its full form (e.g., `pvlib.atmosphere.get_absolute_airmass`)

Reply:
@RDaxini I think there was a slight misunderstanding here -- this comment wasn't about the import style of `get_relative_airmass` but actually about using `get_absolute_airmass` for the pressure correction instead of doing a division with a magic number that some may not recognize.

Reply:
Ah yes, like with the clearsky comments. Understood, thanks. I have made the change.